### PR TITLE
Feature/sample json data

### PR DESCRIPTION
**Pull request by adrian-larysz-rmtcfm-com**

### CODE REVIEW

File: client/src/components/UI/TheResetButton.vue [hunk 1]
```diff
@@ -0,0 +1,42 @@
+<template>
+    <v-btn depressed color="warning" large :loading="loading" @click="handleFlush" style="width: 100%">
```
Comment (daniel-pupczyk-rmtcfm-com) [inline client/src/components/UI/TheResetButton.vue:2]: events at the end, please add eslint

File: client/src/components/UI/TheResetButton.vue [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    },
+
+    methods: {
+        ...mapActions({ reset: 'reset' }),
+        ...mapMutations({ negateRefreshSignal: 'NEGATE_REFRESH_SIGNAL' }),
+
+        async handleFlush() {
+            this.loading = true;
+
+            await this.reset();
```
Comment (daniel-pupczyk-rmtcfm-com) [inline client/src/components/UI/TheResetButton.vue:25]: try-catch

File: client/src/store/index.js [hunk 3]
```diff
@@ -13,24 +13,28 @@ Vue.use(Vuex);
...
+        SET_PERIOD: (state, period) => (state.period = period)
     },
 
     actions: {
         async flush() {
             await axios.delete('/api/flush');
         },
 
+        async reset() {
+            await axios.delete('/api/reset');
```
Comment (daniel-pupczyk-rmtcfm-com) [inline client/src/store/index.js:35]: it shouldnt be `delete` in my opinion, I think it should be POST

File: client/src/store/index.js [hunk 4]
```diff
@@ -13,24 +13,28 @@ Vue.use(Vuex);
...
 
     mutations: {
         NEGATE_REFRESH_SIGNAL: state => (state.refreshSignal = !state.refreshSignal),
-        SET_PERIOD: (state, period) => (state.period = period),
+        SET_PERIOD: (state, period) => (state.period = period)
     },
 
     actions: {
         async flush() {
             await axios.delete('/api/flush');
```
Comment (daniel-pupczyk-rmtcfm-com) [inline client/src/store/index.js:31]: as below

File: server/src/bin/sample.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
+const fs = require('fs');
```
Comment (daniel-pupczyk-rmtcfm-com) [inline server/src/bin/sample.js:1]: move this file to folder `scripts`

File: client/src/store/index.js (hunk 4 above)
Comment (adrian-larysz-rmtcfm-com) [reply client/src/store/index.js:31]: @daniel-pupczyk-rmtcfm-com To be honest, it clears everything from redis, so delete is good here.

File: client/src/store/index.js (hunk 3 above)
Comment (adrian-larysz-rmtcfm-com) [reply client/src/store/index.js:35]: @daniel-pupczyk-rmtcfm-com Ok.

File: client/src/components/UI/TheResetButton.vue (hunk 2 above)
Comment (adrian-larysz-rmtcfm-com) [reply client/src/components/UI/TheResetButton.vue:25]: @daniel-pupczyk-rmtcfm-com Ok.

File: client/src/components/UI/TheResetButton.vue (hunk 1 above)
Comment (adrian-larysz-rmtcfm-com) [reply client/src/components/UI/TheResetButton.vue:2]: @daniel-pupczyk-rmtcfm-com Ok.

File: server/src/services/data/SampleDataService.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+    async generate() {
+        try {
+            await this.redisService.flush();
+
+            const events = require('../../sample.json');
+
+            for (const event of events) {
+                await this.storeDataService.store(event);
+            }
+        } catch (err) {}
```
Comment (rafal-ksiazek-rmtcfm-com) [inline server/src/services/data/SampleDataService.js:16]: console.error(err);

**Review by rafal-ksiazek-rmtcfm-com — APPROVED** [review]

File: server/src/index.js [hunk 7]
```diff
@@ -32,6 +32,12 @@ app.use(errorHandler);
 
 const { port } = config.server;
 
-app.listen(port, () => {
-    console.log(`Server is listening on port ${port}...`);
-});
+di.get('services.data.sample')
```
Comment (rafal-ksiazek-rmtcfm-com) [inline server/src/index.js:35]: do we need this?

Comment (adrian-larysz-rmtcfm-com) [reply server/src/index.js:35]: @rafal-ksiazek-rmtcfm-com Yes, because we want to load all sample data before server starts. Or not?

**Pull request merged**